### ISSUE TITLE
The resolution specified was not found

### ISSUE THREAD

**Issue by ramas-jpg** [simplymemes/crunchyroll-dl#7]
Hello simplymemes, how are you?

I'm trying to download an episode of Boruto in 1080p, and I get the following error: "The resolution specified was not found"

I can download the episode in the "480" and "720p" normally. Any idea what that might be?

I'm from Brazil (pt-br)

`crunchyroll-dl -i https://www.crunchyroll.com/boruto-naruto-next-generations/episode-97-shikadais-decision-780972 -q 1080p -u username -p password -l ptBR -o ":name :ep"`

even with a Brazilian url, the same error occurs:

`crunchyroll-dl -i https://www.crunchyroll.com/pt-br/boruto-naruto-next-generations/episode-97-shikadais-decision-780972 -q 1080p -u username -p password -l ptBR -o ":name :ep"`

**Comment by Noctunus** [comment]
Hey @ramas-jpg, 

I also had this issue quite often in the past and it seems that the mobile api provides resolution on an on-demand basis. When I tried the same url 5 to 10 times it eventually started downloading as the resolution was then available.
I can't really specify after which time or number of repetition it worked but in any cases - when on the site itself the resoultion is available for this episode - it also worked with the downloader after some tries.
But beware! Don't overdo it with logins - a simple loop is written very easy but you'll get blocked by crunchy if you are logging in very often in a short time-frame.

@simplymemes have you also noticed this behavior and if yes, would it be ok for you to implement an automatic retry while still logged in?

**Comment by simplymemes** [comment]
@ramas-jpg For that specific media, I was not able to ever get a 1080p version of it... It may be just because CR doesn't offer it in 1080p version in Portuguese, however they do in English. 

@Noctunus Can you possibly provide an example of said series for me to test?

**Comment by Noctunus** [comment]
@simplymemes Tested a bit with different series - the problem seems to be gone. I was wondering anyway why this is happening at all but that seems to be resolved on the crunchy side.
Nevertheless - if it comes up again as an issue I'll provide an example and open a new ticket for it.

**Comment by LutzPS** [comment]
This https://www.crunchyroll.com/pt-br/mob-psycho-100/episode-1-self-proclaimed-psychic-reigen-arataka-and-mob-776682 as 1080p, but crunchyroll-dl won't find

`crunchyroll-dl --username "EMAIL" --password "PASS" --language ptBR --quality 1080p --dont-autoselect-quality --input https://www.crunchyroll.com/pt-br/mob-psycho-100/episode-1-self-proclaimed-psychic-reigen-arataka-and-mob-776682
i crunchyroll-dl v2.0.0-beta.1
i Attempting to login...
i Successfully logged in!
i Logged in with a premium account.
i Attempting to fetch episode...
i Could not find resolution (1080p) specified, not falling back
i Available resolutions: 240p, 360p, 480p, 720p
i Done!
i Logging out...`

**Comment by simplymemes** [comment]
@LutzPS I only have access to what the app provides, are you able to watch in 1080p in the app?

**Comment by LutzPS** [comment]
> @LutzPS I only have access to what the app provides, are you able to watch in 1080p in the app?

Idk how see in app if is 720p or 1080p.

But, some episode of Black Clover https://www.crunchyroll.com/pt-br/black-clover-international-dubs , crunchyroll-dl get 1080p others 720p

> (Dublado PT) Black Clover Episode 1 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 2 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 3 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 4 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 5 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 6 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 7 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 8 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 9 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 10 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 11 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 12 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 13 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 14 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 15 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 16 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 17 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 18 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 19 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 20 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 21 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 22 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 23 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 24 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 25 [720p] [Hard].mp4
> (Dublado PT) Black Clover Episode 26 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 27 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 28 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 29 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 30 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 31 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 32 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 33 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 34 [1080p] [Hard].mp4
> (Dublado PT) Black Clover Episode 35 [1080p] [Hard].mp4

**Comment by LutzPS** [comment]
Nice, work

> crunchyroll-dl --username "USER" --password "PASS" --language ptBR --quality 1080p --dont-autoselect-quality --vilos -s soft --input https://www.crunchyroll.com/pt-br/mob-psycho-100/episode-1-self-proclaimed-psychic-reigen-arataka-and-mob-776682
> i crunchyroll-dl v2.0.0-beta.2
> i Attempting to login...
> i Successfully logged in!
> i Logged in with a premium account.
> i Attempting to fetch episode...
> √ Which subtitle languages would you like to download? » Português (Brasil)
> i Downloading subtitle languages: Português (Brasil)
> i Subtitles downloaded!
> i Downloading episode as "Mob Psycho 100 (Portuguese Dub) Episode 1 [1080p] [Soft].mp4"
> i Beginning download...
> i Downloading..  100.00% 129/fps 00:24:08.83/encoded
> i Muxing...
> i Successfully downloaded "Mob Psycho 100 (Portuguese Dub) Episode 1 [1080p] [Soft].mp4"
> i Done!
> i Logging out...
